src/mappings/StakingRewards/StakingRewards.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  RewardAdded,
+  RewardPaid,
+  RewardsVaultSet,
+  Staked,
+  Withdrawn,
+} from '../../../generated/templates/StakingRewards/StakingRewards'
+import { StakingRewardsContractType } from '../../enums'
+import {
+  handleRewardAddedForType,
+  handleStakedForType,
+  handleRewardsVaultSetForType,
+  handleRewardPaidForType,
+  handleWithdrawnForType,
+} from './shared'
+
+export function handleRewardAdded(event: RewardAdded): void {
```

Comment:
Need to handle the rewardsDistributor getting updated.. there is no event for that at the mo (my bad). Maybe in `updateStakingRewards` we can simply refresh the `rewardsDistributor` address. I'll think about adding the event to the contracts

Reply:
Ok, I'll add the call there and we can see if that's helpful to have :+1: